pkg/storegateway/indexheader/lazy_binary_reader.go

```diff
@@ -82,6 +82,8 @@ type LazyBinaryReader struct {
 
 	// Keep track of the last time it was used.
 	usedAt *atomic.Int64
+
+	blockID string
```

Comment:
Should we store this as a `ulid.ULID`, and convert it to a string closer to where we need it to be a string?

Reply:
Don't have any preference. I converted to string early because I know I only need string for now. But I can use ulid too.

Reply:
I converted this to `map[ulid.ULID]int641`.